File: Gestion de la Informacion/alumnosP7 (4)/alumnosP7/Provincia.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using BDLibrary;

namespace farmacia
{
    public class Provincia
    {
        private string cod;
        private string nombre;

        private static string BD_SERVER = Properties.Settings.Default.BD_SERVER;
        private static string BD_NAME = Properties.Settings.Default.BD_NAME;

        public static List<Provincia> ListaProvincias()
        {
            List<Provincia> lista = new List<Provincia>();
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
            foreach (object[] tupla in miBD.Select("SELECT Codigo FROM tProvincia;"))
            {
                string id = (string)tupla[0];
                Provincia p = new Provincia(id);
                lista.Add(p);
            }

            return lista;
        }

        public Provincia(string cod)
        {
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
            object[] tupla = miBD.Select("SELECT * FROM tProvincia WHERE Codigo='" + cod + "';")[0];

            this.cod = (string)tupla[0];
            this.nombre = (string)tupla[1];
        }

        public Provincia(string cod, string des)
        {
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
            miBD.Insert("INSERT INTO tProvincia VALUES('" + cod + "', '" + des + "');");
            this.cod = cod;
            this.nombre = des;
        }

        public string Codigo
        {
            get { return cod; }
            set
            {
                SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                miBD.Update("UPDATE tProvincia SET Codigo = '" + value
                    + "' WHERE Codigo='" + this.cod + "';");
                cod = value;
            }
        }

        public string Nombre
        {
            get { return nombre; }
            set
            {
                SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                miBD.Update("UPDATE tProvincia SET Nombre = '" + value
                    + "' WHERE Codigo='" + this.cod + "';");
                nombre = value;
            }
        }

        public void BorrarProvincia()
        {
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
            miBD.Delete("DELETE FROM tProvincia WHERE Codigo='" + this.cod + "';");

            this.cod = null;
            this.nombre = null;
        }

        public override string ToString()
        {
            return nombre;
        }

        public override bool Equals(object obj)
        {
            return obj is Provincia && (((Provincia)obj).Codigo.ToUpper().CompareTo(this.Codigo.ToUpper()) == 0);
        }

        public override int GetHashCode()
        {
            return Codigo.GetHashCode();
        }
    }
}

File: Gestion de la Informacion/Practica3/src/Provincia.java
import java.util.*;

public class Provincia 
{
    private static String BD_SERVER = "localhost";
    private static String BD_NAME = "GI1920";
    
	private String codigo;
	private String nombre;
	
	public static List<Provincia> ListaProvincias()
	{
        		// COMPLETAR
		BD bd = new BD(BD_SERVER, BD_NAME);
		List<Object[]> lista =  bd.Select("SELECT codigo, nombre FROM tProvincia;");
		bd.finalize();
		List<Provincia> listaProvincias = new ArrayList<>();
		
		for(Object[] row : lista) {
			String codigo = (String) row[0];
			String nombre = (String) row[1];
			
			Provincia provincia = new Provincia(codigo, nombre);
			listaProvincias.add(provincia);
		}
		
		return listaProvincias;
	}

	
	public Provincia(String codigo)
	{
        		// COMPLETAR
		BD bd = new BD(BD_SERVER, BD_NAME);
		List<Object[]> Provs= bd.Select("SELECT nombre FROM tProvincia t WHERE t.codigo = '" + codigo + "';");
		this.codigo= codigo;
		this.nombre = Provs.get(0).toString();
		bd.finalize();
	}
	
	public Provincia(String codigo, String provincia)
	{
        		// COMPLETAR
		
		BD bd = new BD(BD_SERVER, BD_NAME);
		List<Object[]> Provs= bd.Select("SELECT * FROM tProvincia t WHERE t.codigo = '" + codigo + "';");
		if(Provs.isEmpty()) {
			bd.Insert("INSERT INTO tProvincia VALUES('"+codigo+ "','"+provincia+"');");
		}
		bd.finalize();
		this.codigo = codigo;
		this.nombre = provincia;
	}
	
	public String getCodigo() 
	{
		// Devuelve el valor del atributo
		return this.codigo;
	}
	
	public void setCodigo(String value) 
	{
		// Establece el valor del atributo y actualiza la tabla
        		// COMPLETAR
		this.codigo = value;
		BD bd = new BD(BD_SERVER, BD_NAME);
    	bd.Update("UPDATE tprovincia set codigo='"+this.codigo+"';");
    	bd.finalize();
	}
	public String getProvincia() 
	{
		// Devuelve el valor del atributo
		return this.nombre;
	}
	
	public void setProvincia(String value) 
	{
		// Establece el valor del atributo y actualiza la tabla
        // COMPLETAR
		this.nombre=value;
		BD bd = new BD(BD_SERVER, BD_NAME);
    	bd.Update("UPDATE tprovincia set nombre='"+this.nombre+"';");
    	bd.finalize();
	}
	
	public void borrarProvincia()
	{
        // COMPLETAR
		BD bd = new BD(BD_SERVER, BD_NAME);
    	bd.Delete("DELETE FROM tProvincia WHERE codigo='"+this.codigo+"');");
    	bd.finalize();
	}
		   
	public String toString()	
	{	        
		return nombre;
	}
	   
	public boolean equals(Object obj)	   
	{	       
		return obj instanceof Provincia 
	       		&& (((Provincia)obj).codigo.compareToIgnoreCase(this.codigo) == 0);	   
	}

	public int hashCode()	   
	{	        
		return codigo.hashCode();	    
	}
}
